akka-actor-typed-tests/src/test/scala/docs/akka/typed/supervision/SupervisionCompileOnly.scala

```diff
@@ -54,4 +55,51 @@ object SupervisionCompileOnly {
   //#top-level
   Behaviors.supervise(counter(1))
   //#top-level
+
+  //#restart-stop-children
+  def child(size: Long): Behavior[String] =
+    Behaviors.receiveMessage(msg ⇒ child(size + msg.length))
+
+  def parent: Behavior[String] = {
+    Behaviors.supervise[String] {
+      Behaviors.setup { ctx ⇒
+        val child1 = ctx.spawn(child(0), "child1")
+        val child2 = ctx.spawn(child(0), "child2")
+
+        Behaviors.receiveMessage[String] { msg ⇒
+          // there might be bugs here...
+          val parts = msg.split(" ")
+          child1 ! parts(0)
+          child2 ! parts(1)
+          Behaviors.same
+        }.receiveSignal {
+          // stop children when restarted, new instances are created from setup
+          case (c, _: PreRestart) ⇒
+            c.children.foreach(c.stop)
```

Comment:
This approach doesn't work as noted in https://github.com/akka/akka/issues/25556#issuecomment-420765175
The name isn't free until receiving `Terminated`, hm that becomes complicated.